code/modules/reagents/chemistry/reagents/other_reagents.dm

```diff
@@ -217,7 +217,8 @@
 /datum/reagent/fuel/unholywater/reaction_mob(mob/living/M, method=TOUCH, reac_volume)
 	if(method == TOUCH || method == VAPOR)
 		M.reagents.add_reagent("unholywater", reac_volume)
-		return ..()
+		return
```

Comment:
if you return here it won't add firestacks like it should, there's no need for it